### PR TITLE
Added files and error handling

### PR DESCRIPTION
**Pull request by Prajval-Aggarwal**
### Description
Handling errors for commonly used functions. Adding new commonly used functions

### Changes Made

- Returning errors from some commonly used function
- Added Function
    - IsPassValid(password string) error
    - HashPassword(password string) (*string, error)

### Related Issues
NA

### CODE REVIEW

**Comment by SalmaanChicmic** [comment]
Add the function signatures in the PR description please.

File: Makefile [hunk 1]
```diff
@@ -17,4 +17,5 @@ lint-fix:
 	@docker-compose --file=$(lint_docker_compose_file) run gin-golinter golangci-lint run --fix \
 		&& echo "✔️  fixed without errors" \
 		|| (echo "⚠️️  you need to fix above issues manually" && exit 1)
-	@echo "⚠️️ run \"make lint-check\" again to check what did not fix yet"
\ No newline at end of file
+	@echo "⚠️️ run \"make lint-check\" again to check what did not fix yet"
```
Comment (SalmaanChicmic) [inline Makefile:20]: What is the change here?

File: server/gateway/middleware.go [hunk 2]
```diff
@@ -75,17 +75,17 @@ func AdminAuthorization(c *gin.Context) {
...
-
-        c.Next()
-    }
-}
\ No newline at end of file
+	return func(c *gin.Context) {
+		c.Writer.Header().Set("Access-Control-Allow-Origin", "*")
+		c.Writer.Header().Set("Access-Control-Allow-Credentials", "true")
+		c.Writer.Header().Set("Access-Control-Allow-Headers", "Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization, accept, origin, Cache-Control, X-Requested-With")
+		c.Writer.Header().Set("Access-Control-Allow-Methods", "*")
```
Comment (SalmaanChicmic) [inline server/gateway/middleware.go:82]: Reason for allowing all methods?

File: Makefile (hunk 1 above)
Comment (Prajval-Aggarwal) [reply Makefile:20]: nothing sir I think it picked up empty space.

File: server/gateway/middleware.go (hunk 2 above)
Comment (Prajval-Aggarwal) [reply server/gateway/middleware.go:82]: Sir, when I was dealing with mattermost , it required all methods allowed. So I changed it in boiler plate also

File: server/request/auth_request.go [hunk 3]
```diff
@@ -1,6 +1,18 @@
...
+import (
+	validation "github.com/go-ozzo/ozzo-validation"
+	"github.com/go-ozzo/ozzo-validation/is"
+)
+
 type AuthRequest struct {
-	Email    string `json:"email" binding:"required,email" example:"user@example.com"`
-	Password string `json:"password" binding:"required" example:"11111111"`
+	Email    string `json:"email"`
+	Password string `json:"password"`
```
Comment (SalmaanChicmic) [inline server/request/auth_request.go:10]: Why have we removed the validations here?

File: server/utils/commonFunc.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	if !hasSpecial {
+		return errors.New("password do not contain any special character")
+	}
+	return nil
+}
+
+func HashPassword(password string) (*string, error) {
+	bs, err := bcrypt.GenerateFromPassword([]byte(password), 10)
```
Comment (SalmaanChicmic) [inline server/utils/commonFunc.go:53]: what is `10` in bcrypt.GenerateFromPassword([]byte(password), 10)

File: server/utils/commonFunc.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
+package utils
+
+import (
+	"errors"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+func IsPassValid(password string) error {
```
Comment (SalmaanChicmic) [inline server/utils/commonFunc.go]: change the function signature to `IsPassValid(password string) (bool,error)`

File: server/utils/commonFunc.go (hunk 4 above)
Comment (Prajval-Aggarwal) [reply server/utils/commonFunc.go:53]: It is the cost for generating the hash. Default is 10 so i have used 10

File: server/utils/commonFunc.go (hunk 5 above)
Comment (Prajval-Aggarwal) [reply server/utils/commonFunc.go]: Done

**Review by SalmaanChicmic — APPROVED** [review]

**Pull request merged**